common/protos/proto/metadata.proto

```diff
@@ -0,0 +1,248 @@
+// There is no guarantee that two get-database request by name will return the same instance.
+message DatabaseNameIdent {
+  uint64 ver = 100;
+
+  // The user this db belongs to
+  string tenant = 1;
+
+  // Database name
+  string db_name = 2;
+}
+
+// The identifier of a database.
+// `db_id` will never change. It can only be created or removed.
+// The `seq` indicates changes of the instance.
+// I.e., the tuple `(db_id, seq)` always identifies the same instance.
+message DatabaseIdent {
+  uint64 ver = 100;
+
+  uint64 db_id = 1;
+
+  // The seq of DatabaseMeta
+  uint64 seq = 2;
+}
+
+// DatabaseMeta is a container of all non-identity information.
+message DatabaseMeta {
+  uint64 ver = 100;
+
+  // Database engine, like github engine.
+  string engine = 5;
+
+  // Database engine options, like github engine <token>.
+  map<string, string> engine_options = 6;
+
+  // Database options.
+  map<string, string> options = 4;
+
+  // The time table created.
+  string created_on = 20;
+
+  // The time table updated.
+  string updated_on = 21;
+
+  // Comment about this table.
+  string comment = 22;
+}
+
+
+// Complete table info.
+message TableInfo {
+  uint64 ver = 100;
+
+  TableIdent ident = 1;
+
+  /// For a table it is `db_name.table_name`.
+  /// For a table function, it is `table_name(args)`
+  string desc = 2;
+
+  /// `name` is meant to be used with table-function.
+  /// Table-function is identified by `name`.
+  /// A table in the contrast, can only be identified by table-id.
+  string name = 3;
+
+  /// The essential information about a table definition.
+  ///
+  /// It is about what a table actually is.
+  /// `name`, `id` or `version` is not included in the table structure definition.
+  TableMeta meta = 4;
+}
+
+// The identifier of a table by name. Names can be changed.
+// There is no guarantee that two get-table request by name will return the same instance.
+message TableNameIdent {
+  uint64 ver = 100;
+
+  // The user this table belongs to.
+  string tenant = 1;
+
+  // The database this table belongs to.
+  string db_name = 2;
+
+  // The table name.
+  string table_name = 3;
+}
+
+// The identifier of a table.
+// `table_id` will never change. It can only be created or removed.
+// The `seq` indicates changes of the instance.
+// I.e., the tuple `(db_id, seq)` always identifies the same instance.
+message TableIdent {
+  uint64 ver = 100;
+
+  uint64 table_id = 1;
+
+  // The seq of TableMeta
+  uint64 seq = 2;
+}
+
+// TableMeta is a container of all non-identity information.
+message TableMeta {
+  uint64 ver = 100;
+
+  // Table schema, i.e., columns definition.
+  DataSchema schema = 1;
+
+  // Table engine.
+  string engine = 6;
+
+  // Database engine options, like github engine <token>.
+  map<string, string> engine_options = 7;
+
+  // Table options.
+  map<string, string> options = 5;
+
+  // Keys to sort rows in table.
+  optional bytes order_keys = 9;
+
+  // The time table created.
+  string created_on = 20;
```

Comment:
it would be better to add 'updated' and 'comment' for table like database.

Reply:
Ok